### PR TITLE
Support non-libffi builds without caching ForeignFunctions examples

### PR DESCRIPTION
**Pull request by d-torrance**
This partially reverts #2672.  Rather than still shipping a non-working copy of the `ForeignFunctions` package with cached examples when Macaulay2 is built with `--without-libffi` (autotools) or `-DWITH_FFI=OFF)` (cmake), we remove `ForeignFunctions` from the list of distributed packages.

This is a draft PR for now.  The last commit sets the GitHub builds to build without libffi.  If that goes well, then I'll remove that commit.

### CODE REVIEW

**Comment by mikestillman** [comment]
@d-torrance Which pull request is planning on going into 1.21?  I'll try that one on my mac.  I tried one earlier, and was down to one example error, rather than 5.

**Comment by d-torrance** [comment]
> @d-torrance Which pull request is planning on going into 1.21? I'll try that one on my mac. I tried one earlier, and was down to one example error, rather than 5.

Both #2705 and #2707.  #2705 fixes 4 of the example errors, and #2707 "fixes" the 5th one by removing that feature for now.

Would it be easier if it was just one pull request?  The problems weren't closely related so I kept them separate.

**Comment by mikestillman** [comment]
Seems to work now on my apple M1 !  Thanks @d-torrance!  @DanGrayson I think I'm done with changes for 1.21...

**Comment by d-torrance** [comment]
The macOS autotools build for this PR was failing because of a subtle difference between GNU sed and BSD sed.  Just force-pushed a fix.

**Comment by d-torrance** [comment]
I redid this PR based on @mahrud's [comment](https://github.com/Macaulay2/M2/pull/2707#issuecomment-1336040786).   If Macaulay2 is built w/o libffi support, then just a tiny stub package is loaded with a single documentation page that says that the package doesn't work.

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -134,7 +134,8 @@ jobs:
             -DCMAKE_PREFIX_PATH="`brew --prefix`;`brew --prefix libffi`" \
             -DCMAKE_INSTALL_PREFIX=/usr \
             -DWITH_PYTHON=ON \
-            --debug-trycompile
+            --debug-trycompile \
+            -DWITH_FFI=OFF
```
Comment (mahrud) [inline .github/workflows/test_build.yml:138]: Why does CMake build without FFI?

Comment (d-torrance) [reply .github/workflows/test_build.yml:138]: This was from a temporary commit to check and make sure that the various builds all worked when disabling libffi.  They did, so I'll remove it here in a moment.

Comment (mahrud) [reply .github/workflows/test_build.yml]: Ah I see.

**Pull request merged**